### PR TITLE
(Feature) GoodSwap should have generic analytics sdk

### PR DESCRIPTION
**Pull request by johnsmith-gooddollar**
# Description

- similar to gooddapp, send same event to multiple analytics services
- GA, Amplitude, Sentry and indicative.com included
- added react hooks & context provider

About #239

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/sdk-v2/src/sdk/analytics/react.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, { Children, createContext, ReactNode, useCallback, useContext, useEffect, useRef, useState } from 'react'
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/react.tsx:1]:
```suggestion
import React, { createContext, ReactNode, useCallback, useContext, useEffect, useRef, useState } from 'react'
```

File: packages/sdk-v2/src/sdk/analytics/react.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  children?: ReactNode;
+}
+
+export const AnalyticsContext = createContext<IAnalyticsContext>({
+  identify: (identifier: string | number, email?: string, props?: object): void => {},
+  send: (event: string, data?: object): void => {},
+  capture: (exception: Error, fingerprint?: string[], tags?: object, extra?: object): void => {},
+})
+
+export function AnalyticsProvider({ config, appProps }: IAnaliticsProviderProps): ReactNode | null {
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/react.tsx:19]:
```suggestion
export function AnalyticsProvider({ config, appProps, children }: IAnaliticsProviderProps): ReactNode | null {
```

File: packages/sdk-v2/src/sdk/analytics/react.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  useEffect(() => {
+    const sdk = new Analytics(configRef.current)
+
+    sdk.initialize(appPropsRef.current).then(() => setSDK(sdk))
+  }, [setSDK])
+
+  return !sdk ? null : (
+    <AnalyticsContext.Provider value={{ send, capture, identify }}>
+      {Children}
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/react.tsx:44]:
```suggestion
      {children}
```

**Comment by roma-hladilka** [comment]
@johnsmith-gooddollar 
as I can see, we have types folder in gitignore for sdk-v2, but it was pushed to the repo. Do we need to delete it?

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts [hunk 4]
```diff
@@ -1,48 +1,50 @@
...
-    })
+      }
+    });
   }
 }
 
-export default new Wrapper(Amplitude.getInstance(), Identify)
+const wrapper: Wrapper = new Wrapper(Amplitude.getInstance(), Identify);
+
+export default wrapper;
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts:50]: revert this. just 

```typescript
new Wrapper(Amplitude.getInstance(), Identify);
```

@L03TJ3 had to solve (or will solve) issues with autogenerating indexes. As i remember we agreed don't use ctix for now. Just exclude ctix calls from package.json scripts:

```json
    "build": "yarn dev:clean && tsc",
    "build:release": "yarn dev:clean && tsc -p tsconfig.release.json",
    "dev:clean": "rm -fr dist types tsconfig.tsbuildinfo",
```

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
-import amplitude from 'amplitude-js'
+import amplitude from "amplitude-js";
 
-export default amplitude.getInstance()
+const amplitudeInstance: amplitude.AmplitudeClient = amplitude.getInstance();
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/amplitude/api.ts:3]: see above

File: packages/sdk-v2/src/sdk/analytics/google/types.ts [hunk 6]
```diff
@@ -11,6 +11,6 @@ export interface IGoogleAPI {
   logEvent(event: string, data?: any): void;
 }
 
-export const defaultConfig: Partial<IGoogleConfig> = {
-  userProperty: 'user'
-}
+export const defaultGoogleConfig: Partial<IGoogleConfig> = {
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/google/types.ts:14]: revert to `defaultConfig`. just do not re-export it in 'index.ts' barrel. it's for internal usage only

File: packages/sdk-v2/src/sdk/analytics/index.tsx [hunk 7]
```diff
@@ -1,7 +0,0 @@
-export * from './amplitude'
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/index.tsx]: pls revert all indexes autogenerated by ctix. rename this index.tsx to ts manually. adjust/fix exports manually. do not re-rexport 'defaultConfig' here

**Review by johnsmith-gooddollar — COMMENTED** [review]
@roma-hladilka ^^^

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts (hunk 4 above)
Comment (roma-hladilka) [reply packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts]: reverted, adjusted scritps

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.ts (hunk 5 above)
Comment (roma-hladilka) [reply packages/sdk-v2/src/sdk/analytics/amplitude/api.ts]: done

File: packages/sdk-v2/src/sdk/analytics/google/types.ts (hunk 6 above)
Comment (roma-hladilka) [reply packages/sdk-v2/src/sdk/analytics/google/types.ts]: done

File: packages/sdk-v2/src/sdk/analytics/index.tsx (hunk 7 above)
Comment (roma-hladilka) [reply packages/sdk-v2/src/sdk/analytics/index.tsx]: reverted, adjusted

File: packages/sdk-v2/package.json [hunk 8]
```diff
@@ -4,9 +4,9 @@
   "version": "0.1.35",
   "description": "ethers and react hooks based on usedapp sdk for GoodDollar protocol",
   "scripts": {
-    "build": "yarn dev:clean && ctix c && tsc",
-    "build:release": "yarn dev:clean && ctix c && tsc -p tsconfig.release.json",
-    "dev:clean": "rm -fr dist types tsconfig.tsbuildinfo && ctix r",
+    "build": "yarn dev:clean && tsc",
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/package.json:7]: this should be revert this before merge

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts [hunk 9]
```diff
@@ -28,11 +28,11 @@ class Wrapper {
     this.api.setUserId(id);
   }
 
-  setUserProperties(props: object) {
+  setUserProperties(props: any) {
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/amplitude/api.native.ts:31]: why 'any' ? if object doesn't works, could we use something like key-value dictionary (e.g. `Record<string, any>`) ?

File: packages/sdk-v2/src/sdk/analytics/amplitude/index.ts [hunk 10]
```diff
@@ -1,2 +1,2 @@
 export * from "./Amplitude";
-export { IAmplitudeConfig } from "./types";
+export type { IAmplitudeConfig } from "./types";
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/amplitude/index.ts:2]: revert. this isn't flow, there's no `export type`, just `export`

**Review by johnsmith-gooddollar — COMMENTED** [review]
^^^ @alt-a1r

File: packages/sdk-v2/src/sdk/analytics/amplitude/index.ts (hunk 10 above)
Comment (alt-a1r) [reply packages/sdk-v2/src/sdk/analytics/amplitude/index.ts:2]: it throws a compile error on build when there are just export

File: packages/sdk-v2/src/sdk/analytics/amplitude/api.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+  async initialize(apiKey: string) {
+    return this.api.init(apiKey);
+  }
+
+  setUserId(id: string): void {
+    this.api.setUserId(id);
+  }
+
+  setUserProperties(props: Record<string, unknown>) {
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/analytics/amplitude/api.ts:15]: 1. `Record<string, any>`, not `unknown`
2. you  still have `object` in many places

**Review by johnsmith-gooddollar — COMMENTED** [review]
^^^

**Review by johnsmith-gooddollar — COMMENTED** [review]
approved

**Pull request merged**